### ISSUE TITLE
cylindrical var coordinate system is broken

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#159]
The coordinate system cylindrical var, which was supposed to allow for a variable height to the accretion disk is broken, with errors below.  It does not appear to be asking for the variables that would describe how the the disk height varies with radius

[sakana:41842] **\* Process received signal ***
[sakana:41842] Signal: Segmentation fault: 11 (11)
[sakana:41842] Signal code: Address not mapped (1)
[sakana:41842] Failing at address: 0x131f83070
[sakana:41842] [ 0] 2   libsystem_platform.dylib            0x00007fff849ed5aa _sigtramp + 26
[sakana:41842] [ 1] 3   ???                                 0x00007fff5f2af180 0x0 + 140734790037888
[sakana:41842] [ 2] 4   py                                  0x000000010095d20b define_wind + 267
[sakana:41842] [ 3] 5   py                                  0x0000000100950d24 main + 3876
[sakana:41842] [ 4] 6   libdyld.dylib                       0x00007fff8e6bb5fd start + 1
[sakana:41842] [ 5] 7   ???                                 0x0000000000000002 0x0 + 2
[sakana:41842] **\* End of error message ***

**Comment by kslong** [comment]
program is dying in cylvar_make_grid.  (Note that it does not matter whether one says the disk is extended in the vertical direction to produce this error)

**Comment by kslong** [comment]
Program is failing in line 174 of this routine after about 877 cycles through the loop.  It looks nominal prior to this.  MDIM and NDIM are both set to 30, so it is unclear why this is happening.   I cannot print out the geo parameters in ddd for some reason, but the values in the loop look quite nominal.

**Comment by kslong** [comment]
Note that the program fails in different places for me now, on my Mac and on linux.  It gets considerable furether on linux.  (The problem is in the dev branch and the domain2 branch)

(Note - on my Mac, despite efforts to use gcc, I am really compiling with clang)

**Comment by kslong** [comment]
We still need to have a good example that we check for this coordinate system.  It should probably be some kind of disk system with disk set so it absorbs photons and changes the temperature of the disk, so it can be compared to a flat disk.

**Comment by kslong** [comment]
The main problem here actually had to do with finding the disk boundary when one had a vertically extended disk.

- In some case, the routine that found the disk had an error in finding the boundary that was quite small, basically a round-off error.  When we checked whether we inside or outside the disk, we got an error because of the roundoff error.
-Additionally, there was an inconsistency in the way a vertically extended disk was defined.  In dick.c the outer edge was defined as a sphere of radius geo.diskmax, whereas else where it was defined as a cylinder.  When this error was removed most of the errors went away.  

The .pf file  used to test for the above errors was 

[flat.pf.txt](https://github.com/agnwinds/python/files/2897384/flat.pf.txt)

After this was corrected, I was able to calculate the spectrum of a flared disk with or without cylin_var coords. There were occasional errors clearly associated with the coordinate system, but the spectra were very similar to those produced with pure cylindrical coordinates.   While it's clear that more tests should be made, if we make extensive use of cylind_var coordinates, I am going to close this issue.
